### PR TITLE
fix(plasma-ui): Fix scrollable behavior for `Sheet` component

### PR DESCRIPTION
**Pull request by neretin-trike**
Исправил поведение компонента `Sheet` для случаев, когда внутри главного контейнера есть ещё дочерние контейнеры, которые имеют скролл.

<details>
  <summary>:baby_chick: Download canary assets:</summary>
  <blockquote>
    <details>
      <summary>color</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_ios-swift--canary.606.5542859479.swift'>color_asdk_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_kotlin--canary.606.5542859479.kt'>color_asdk_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_react-native--canary.606.5542859479.ts'>color_asdk_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_xml--canary.606.5542859479.xml'>color_asdk_xml--canary.606.5542859479.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_ios-swift--canary.606.5542859479.swift'>color_plasma_b2c_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_kotlin--canary.606.5542859479.kt'>color_plasma_b2c_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_react-native--canary.606.5542859479.ts'>color_plasma_b2c_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_xml--canary.606.5542859479.xml'>color_plasma_b2c_xml--canary.606.5542859479.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.606.5542859479.swift'>color_sberHealth_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.606.5542859479.kt'>color_sberHealth_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.606.5542859479.ts'>color_sberHealth_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_xml--canary.606.5542859479.xml'>color_sberHealth_xml--canary.606.5542859479.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.606.5542859479.swift'>color_sbermarket_business_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.606.5542859479.kt'>color_sbermarket_business_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.606.5542859479.ts'>color_sbermarket_business_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_xml--canary.606.5542859479.xml'>color_sbermarket_business_xml--canary.606.5542859479.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.606.5542859479.swift'>color_sbermarket_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.606.5542859479.kt'>color_sbermarket_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.606.5542859479.swift'>color_sbermarket_metro_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.606.5542859479.kt'>color_sbermarket_metro_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.606.5542859479.ts'>color_sbermarket_metro_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_xml--canary.606.5542859479.xml'>color_sbermarket_metro_xml--canary.606.5542859479.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.606.5542859479.ts'>color_sbermarket_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.606.5542859479.swift'>color_sbermarket_selgros_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.606.5542859479.kt'>color_sbermarket_selgros_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.606.5542859479.ts'>color_sbermarket_selgros_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_xml--canary.606.5542859479.xml'>color_sbermarket_selgros_xml--canary.606.5542859479.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.606.5542859479.swift'>color_sbermarket_wlbusiness_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.606.5542859479.kt'>color_sbermarket_wlbusiness_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.606.5542859479.ts'>color_sbermarket_wlbusiness_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_xml--canary.606.5542859479.xml'>color_sbermarket_wlbusiness_xml--canary.606.5542859479.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_xml--canary.606.5542859479.xml'>color_sbermarket_xml--canary.606.5542859479.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_ios-swift--canary.606.5542859479.swift'>color_sberonline_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_kotlin--canary.606.5542859479.kt'>color_sberonline_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_react-native--canary.606.5542859479.ts'>color_sberonline_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_xml--canary.606.5542859479.xml'>color_sberonline_xml--canary.606.5542859479.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.606.5542859479.swift'>color_sberprime_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.606.5542859479.kt'>color_sberprime_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.606.5542859479.ts'>color_sberprime_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_xml--canary.606.5542859479.xml'>color_sberprime_xml--canary.606.5542859479.xml</a><br>
      </blockquote>
    </details>
    <details>
      <summary>shadow</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.606.5542859479.ts'>shadow_sbermarket_react-native--canary.606.5542859479.ts</a><br>
      </blockquote>
    </details>
    <details>
      <summary>typo</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.606.5542859479.swift'>typo_mage_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.606.5542859479.kt'>typo_mage_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.606.5542859479.ts'>typo_mage_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.606.5542859479.swift'>typo_plasma_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.606.5542859479.kt'>typo_plasma_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.606.5542859479.ts'>typo_plasma_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.606.5542859479.swift'>typo_ruler_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.606.5542859479.kt'>typo_ruler_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.606.5542859479.ts'>typo_ruler_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.606.5542859479.swift'>typo_sage_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.606.5542859479.kt'>typo_sage_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.606.5542859479.ts'>typo_sage_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.606.5542859479.swift'>typo_sbermarket_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.606.5542859479.kt'>typo_sbermarket_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.606.5542859479.ts'>typo_sbermarket_react-native--canary.606.5542859479.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.606.5542859479.swift'>typo_soulmate_ios-swift--canary.606.5542859479.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.606.5542859479.kt'>typo_soulmate_kotlin--canary.606.5542859479.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.606.5542859479.ts'>typo_soulmate_react-native--canary.606.5542859479.ts</a><br>
      </blockquote>
    </details>
  </blockquote>
</details>

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.218.1-canary.606.5542859479.0
  npm install @salutejs/plasma-core@1.123.1-canary.606.5542859479.0
  npm install @salutejs/plasma-hope@1.218.1-canary.606.5542859479.0
  npm install @salutejs/plasma-icons@1.152.1-canary.606.5542859479.0
  npm install @salutejs/plasma-temple@1.167.1-canary.606.5542859479.0
  npm install @salutejs/plasma-ui@1.199.1-canary.606.5542859479.0
  npm install @salutejs/plasma-web@1.218.1-canary.606.5542859479.0
  npm install @salutejs/plasma-sb-utils@0.121.1-canary.606.5542859479.0
  # or 
  yarn add @salutejs/plasma-b2c@1.218.1-canary.606.5542859479.0
  yarn add @salutejs/plasma-core@1.123.1-canary.606.5542859479.0
  yarn add @salutejs/plasma-hope@1.218.1-canary.606.5542859479.0
  yarn add @salutejs/plasma-icons@1.152.1-canary.606.5542859479.0
  yarn add @salutejs/plasma-temple@1.167.1-canary.606.5542859479.0
  yarn add @salutejs/plasma-ui@1.199.1-canary.606.5542859479.0
  yarn add @salutejs/plasma-web@1.218.1-canary.606.5542859479.0
  yarn add @salutejs/plasma-sb-utils@0.121.1-canary.606.5542859479.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-606/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-606/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-606/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-606/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-606/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-606/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-606/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: packages/plasma-ui/src/components/Sheet/Sheet.tsx [hunk 1]
```diff
@@ -28,6 +28,11 @@ export interface SheetProps extends React.HTMLAttributes<HTMLDivElement> {
      * Наличие состояния анимации/перехода, по-умолчанию true.
      */
     withTransition?: boolean;
+
+    /**
+     * Throttling внутренних обработчиков события onScroll
+     */
+    throttleMs?: number;
```
Comment (Yakutoc) [inline packages/plasma-ui/src/components/Sheet/Sheet.tsx:35]: [Вкусовщина]

@neretin-trike Про нейминг, почему именно `ms` а не например `scroll`

File: packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts [hunk 2]
```diff
@@ -1,28 +1,48 @@
...
+
+const isScrollable = (element: HTMLElement | null) => {
+    if (!element) {
+        return false;
+    }
+
+    const style = getComputedStyle(element);
+
+    return (
+        style.overflow === 'scroll' ||
```
Comment (Yakutoc) [inline packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:15]: @neretin-trike А мы можем это все на includes переписать?

```js
['scroll', 'auto', 'etc'].includes(item)
```

File: packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts [hunk 3]
```diff
@@ -1,28 +1,48 @@
...
 
-        const hasScroll = contentEl && contentEl.scrollHeight > contentEl.clientHeight;
-        const triggerElement = !isTopScroll && hasScroll ? handleEl : contentWrapperEl;
+        const triggerElement = !isTopScroll && isScrollable(contentEl) ? handleEl : contentWrapperEl;
 
         if (!triggerElement || !contentEl || !contentWrapperEl) {
             return;
         }
 
+        const nodes = Array.from<HTMLElement>(contentWrapperEl.querySelectorAll('*'));
```
Comment (Yakutoc) [inline packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:43]: [Уточнение]

А это `querySelectorAll('*')` не ударит по перфу?

File: packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts (hunk 2 above)
Comment (Yeti-or) [reply packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:15]: А зачем ?

File: packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts (hunk 3 above)
Comment (Yeti-or) [inline packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:43]: почему тут не contentEl ?

File: packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts (hunk 2 above)
Comment (neretin-trike) [reply packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:15]: тут как будто бы в этом смысла особого не будет. Пушто иначе придётся либо пересечение массивов делать (overflowX, overflow) и (scroll,auto), что кажется дороже, чем точечно проверить. А расширяться эта проверка точно не будет

File: packages/plasma-ui/src/components/Sheet/Sheet.tsx (hunk 1 above)
Comment (neretin-trike) [reply packages/plasma-ui/src/components/Sheet/Sheet.tsx:35]: тут я просто взял свойство с аналогичным значением из компонента `Carousel`. В целом мне показалось это нормальным названием

File: packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts (hunk 3 above)
Comment (neretin-trike) [reply packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:43]: ну зависит того, что будут укладывать в `Sheet`. Тут хз как себе подстелить соломку, т.к. мне нужно смотреть все тэги, которые могут скролл (а таких очень много). Но как я понял, работать это должно быстро, по крайней мере в типовых случаях

File: packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts [hunk 4]
```diff
@@ -1,28 +1,48 @@
...
+        const scrollableElements = nodes.filter(isScrollable);
```
Comment (Yeti-or) [inline packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:44]: мы не хотим тут opt-in поставить ?

**Review by Yakutoc — APPROVED** [review]

File: packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts (hunk 3 above)
Comment (neretin-trike) [reply packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:43]: хороший вопрос. Попробую с contentEl

File: packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts (hunk 4 above)
Comment (neretin-trike) [reply packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:44]: не совсем понял, о чём ты

Comment (Yeti-or) [reply packages/plasma-ui/src/components/Sheet/useSheetSwipe.ts:44]: я имею ввиду включить такое поведение по умолчанию но дать возможность отключить проверку во всех скроллобах элементах ? ну типа чтобы не ходить по дому а ля дорого 

хотя кажется что можно и не давать такой возможности давай так оставим

**Review by kayman233 — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-606/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Review by Yeti-or — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-606/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-606/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!

  website: http://plasma.sberdevices.ru/pr/pr-606/
ui storybook: http://plasma.sberdevices.ru/pr/pr-606/ui-storybook/
temple storybook: http://plasma.sberdevices.ru/pr/pr-606/temple-storybook/

**Pull request merged**